consumer/consumererror/combineerrors.go

```diff
@@ -0,0 +1,47 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import (
+	"fmt"
+	"strings"
+)
+
+// CombineErrors converts a list of errors into one error.
+func CombineErrors(errs []error) error {
```

Comment:
Consider to split this change (moving the `CombineErrors`) in a separate PR

Reply:
I left `componenterror.CombineErrors` unchanged and added a note about the necessity of it's deprecation.

I added separate `consumer/consumererror.CombineErrors` only because of dependency cycle. I couldn't use `receiver/scraperhelper.CombineScrapeErrors` function in `consumer/consumererror.ScrapeErrors.Combine()` method because `receiver/scraperhelper` already imports `component/componenterror` package.

Reply:
sure I understand the reason and did not say to not do it, I was just trying to offer a solution do split the PR so we can have an easy review.